R/migrations.R

```diff
@@ -0,0 +1,93 @@
+run_migration <- function(queue, output_dir, from_version, to_version,
+                          dry_run = TRUE) {
+  output_dir <- normalizePath(output_dir, mustWork = TRUE)
+  tasks <- queue$queue$task_list()
+  status <- queue$queue$task_status(tasks)
+  completed_tasks <- tasks[status == "COMPLETE"]
+  migrations <- lapply(completed_tasks, migrate_task, queue,
+                       from_version, to_version, dry_run)
+  summary <- lapply(migrations, function(migration) {
+    list(
+      id = migration$id,
+      action = migration$action
+    )
+  })
+  summary <- do.call(rbind, summary)
+  summary_path <- file.path(output_dir, "summary.csv")
+  message(sprintf("Saving summary csv %s", summary_path))
+  output_path <- file.path(output_dir, "output.qs")
+  utils::write.csv(summary, summary_path, row.names = FALSE)
+  message(sprintf("Saving output qs %s", output_path))
+  qs::qsave(migrations, output_path, preset = "fast")
+  migrations
+}
+
+migrate_task <- function(task_id, queue, from_version, to_version, dry_run) {
+  message(sprintf("Migrating %s", task_id))
+  res <- queue$queue$task_result(task_id)
+  if (!naomi:::is_hintr_output(res) ||
+      !all(c("plot_data_path", "model_output_path") %in% names(res))) {
```

Comment:
Would this happen if an output had partially failed, or if it was a really old task?

Reply:
If it's a really old task, at one point we had different names in the struct.